src/server.rs

```diff
@@ -475,11 +475,26 @@ pub trait Output {
     }
 
     fn notify(&self, message: &str) {
-        let output = serde_json::to_string(
-            &NotificationMessage::new(message.to_owned(), ())
-        ).unwrap();
+        let output = serde_json::to_string(&NotificationMessage::new(message.to_owned(), ()))
+            .unwrap();
         self.response(output);
     }
+
+    fn notify_begin(&self) {
+        self.notify("rustDocument/diagnosticsBegin");
+    }
+
+    fn notify_end(&self) {
+        self.notify("rustDocument/diagnosticsEnd");
+    }
+
+    fn notify_biuld_results(&self,
+                            notifications: &Vec<NotificationMessage<PublishRustDiagnosticsParams>>) {
+        for notification in notifications {
```

Comment:
I was trying to do that, but found the `#[serde(deserialize_with = "url_serde::deserialize", serialize_with = "url_serde::serialize")]` so I didn't know what exactly was important to extract out from the object, ( in a sense that I just get a string at the output trait )